packages/resizable-mixin/resizable-mixin.ts

```diff
@@ -0,0 +1,68 @@
+import { LitElement } from 'lit-element';
+import getResizeObserver from './resize-observer-polyfill';
+import { ResizableClass } from './resizable-class';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+const resizables = new Set();
+
+export const ResizableMixin = <T extends Constructor<LitElement>>(
+  base: T
+): T & Constructor<ResizableClass> => {
+  class Resizable extends base {
+    protected static _resizeObserver?: ResizeObserver;
+
+    connectedCallback() {
+      super.connectedCallback();
+
+      resizables.add(this);
+
+      this._initResizeObserver().then(() => {
+        const observer = Resizable._resizeObserver;
```

Comment:
Since this block is run async, should we check that the element is still connected?

Reply:
Good catch. Fixed, added a test to check for this edge case.

Reply:
nit: there's an edge case for the edge case :) If the component is wrapped inside another wrapper element. Then the wrapper can get disconnected while `this.parentNode` still returns truthy.

Reply:
Thanks, changed to use `isConnected` and added a test to cover that.